src/Chapter2.hs

```diff
@@ -727,7 +741,9 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+-- smartReplicate [] = []
+-- smartReplicate (x:xs) = replicate x x ++ smartReplicate xs
+smartReplicate l = concat (map (\x -> replicate x x) l)
```

Comment:
Great!
If you to write this function even more elegantly, you can use the standard function `concatMap` which is a combination of `concat` and `map` as the name suggests 🙂
After that, you could notice, that you would be able to eta-reduce on the last argument 👍🏼